releasenotes/notes/steppable-optimizers-9d9b48ba78bd58bb.yaml

```diff
@@ -0,0 +1,51 @@
+---
+features:
+  - |
+    Created subclass SteppableOptimizer. The optimizers of this kind will have a step mehtod that can
+    be called instead of performing a full optimization. These optimizers will also have access to
+    the ask and tell interface so that complex experiments can be handled more easily. For more
+    information about this interface see: `ask and tell interface
+    <https://optuna.readthedocs.io/en/stable/tutorial/20_recipes/009_ask_and_tell.html>`_.
+    For example:
+
```

Comment:
Could you add an explanation to this code snippet? And ideally add a first example before this one which is just a very simple use like
```python
for _ in range(maxiter):
     state = optimizer.state
     # readout some stuff
     optimizer.step()
```
This would maybe prepare readers for the next big chunk of code that's incoming 😉